receiver/bindplaneauditlogs/logs.go

```diff
@@ -0,0 +1,226 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bindplaneauditlogs
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"sort"
+	"strings"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.uber.org/zap"
+)
+
+const (
+	bindplaneTimeFormat = time.RFC3339
+)
+
+// Add a wrapper struct for the API response
+type apiResponse struct {
+	AuditEvents []AuditLogEvent `json:"auditEvents"`
+}
+
+type bindplaneAuditLogsReceiver struct {
+	cfg           Config
+	client        httpClient
+	consumer      consumer.Logs
+	logger        *zap.Logger
+	cancel        context.CancelFunc
+	wg            *sync.WaitGroup
+	lastTimestamp time.Time
+}
+
+type httpClient interface {
+	Do(req *http.Request) (*http.Response, error)
+	CloseIdleConnections()
+}
+
+// newBindplaneAuditLogsReceiver returns a newly configured bindplaneAuditLogsReceiver
+func newBindplaneAuditLogsReceiver(cfg *Config, logger *zap.Logger, consumer consumer.Logs) (*bindplaneAuditLogsReceiver, error) {
+	return &bindplaneAuditLogsReceiver{
+		cfg:           *cfg,
+		client:        http.DefaultClient,
+		consumer:      consumer,
+		logger:        logger,
+		wg:            &sync.WaitGroup{},
+		lastTimestamp: time.Now().UTC().Add(-cfg.PollInterval),
+	}, nil
+}
+
+func (r *bindplaneAuditLogsReceiver) Start(_ context.Context, _ component.Host) error {
+	ctx, cancel := context.WithCancel(context.Background())
+	r.cancel = cancel
+	r.wg.Add(1)
+	go r.startPolling(ctx)
+	return nil
+}
+
+func (r *bindplaneAuditLogsReceiver) startPolling(ctx context.Context) {
+	defer r.wg.Done()
+	t := time.NewTicker(r.cfg.PollInterval)
+
+	err := r.poll(ctx)
+	if err != nil {
+		r.logger.Error("there was an error during the first poll", zap.Error(err))
+	}
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-t.C:
+			err := r.poll(ctx)
+			if err != nil {
+				r.logger.Error("there was an error during the poll", zap.Error(err))
+			}
+		}
+	}
+}
+
+func (r *bindplaneAuditLogsReceiver) poll(ctx context.Context) error {
+	logEvents := r.getLogs(ctx)
+	observedTime := pcommon.NewTimestampFromTime(time.Now())
+	logs := r.processLogEvents(observedTime, logEvents)
+	if logs.LogRecordCount() > 0 {
+		if err := r.consumer.ConsumeLogs(ctx, logs); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *bindplaneAuditLogsReceiver) getLogs(ctx context.Context) []AuditLogEvent {
+	var logs []AuditLogEvent
+
+	reqURL := r.cfg.BindplaneURL
+	reqURL = strings.TrimRight(reqURL, "/")
+	if !strings.Contains(reqURL, "://") {
+		reqURL = "http://" + reqURL
+	}
+	reqURL = reqURL + "/v1/audit-events"
```

Comment:
I think it would be better if r.cfg.BindplaneURL was type `url.URL`. Then you can avoid trimming the path because the URL type's methods handle it for you.

```go
reqURL := r.cfg.BindPlaneURL
reqURL.Path = "/v1/audit-events"
...
...
req, err := http.NewRequestWithContext(ctx, "GET", reqURL.String(), nil)
```

This way you do not need to handle edge cases that should have been caught during start validation. Also, sometimes URL's can have funny characters (e.g. IPv6).